hera_cal/vis_clean.py

```diff
@@ -664,10 +786,88 @@ def fourier_filter(self, filter_centers, filter_half_widths, mode,
                         w, _ = zeropad_array(w, zeropad=zeropad[m], axis=m)
                         xp[m] = np.hstack([x[m].min() - (np.arange(zeropad[m])[::-1] + 1) * np.mean(np.diff(x[m])),
                                            x[m], x[m].max() + (1 + np.arange(zeropad[m])) * np.mean(np.diff(x[m]))])
-            mdl, res, info = dspec.fourier_filter(x=xp, data=d, wgts=w, filter_centers=filter_centers,
+            mdl, res = np.zeros_like(d), np.zeros_like(d)
+            if skip_flagged_edge_freqs:
```

Comment:
FYI: the next ~80 lines of code have zero comments, so its hard to understand what you are doing. can you just add some comments at each high-level block to help comprehension? thanks. Also, this is a serious amount of code for `fourier_filter`, which should just be in effect a shallow wrapper for `dspec.fourier_filter`. is it worth making this its own function, both for modularity, but also for code comprehension here? lmk your thoughts thanks

Reply:
I've added more documentation.

As far as breakout functions go. I'm not sure if there is a lot of utility outside of `fourier_filter` within `hera_cal`. I am open to moving them to a breakout function if you think that would help with readability (can see this going the other way though since it requires someone to scroll around to all these different functions to figure out whats going on). 

The ideal place to have all these features might actually be in `uvtools` since they help a lot with the stability of the linear filters and it would be useful for them to be available as defaults for people that are not using them on HERA.